Add dropped attribute count to SpanData, Link and Event

MikeGoldsmith opened the pull request:
Updates #993.

Adds `getDroppedAttributeCount` to Link and Event interfaces and to SpanData class (including nested Link and Event classes) and MessageEvent class.

Not sure if we need it on [TimedEvent](https://github.com/open-telemetry/opentelemetry-java/blob/ce1cd9c3bd740017e72e1c6896688e4031543113/sdk/src/main/java/io/opentelemetry/sdk/trace/TimedEvent.java#L28) as it doesn't implement the `Event` interface - let me know and I can add it.

jkwatson commented on api/src/main/java/io/opentelemetry/trace/Event.java:
```diff
@@ -41,4 +41,11 @@
+   *
+   * @return the number of dropped attributes.
+   */
+  long getDroppedAttributeCount();
```
An int should definitely be sufficient, I would think, to hold the number of dropped attributes.

MikeGoldsmith replied:
Yeah, not sure why I picked long. I'll update to int.

jkwatson commented:
Interesting that we don't even allow user-specified attributes on the basic `Event`/`MessageEvent`, so I'm not sure why we need them at all.  @bogdandrutu ?

jkwatson commented:
@MikeGoldsmith looks like this needs a rebase

carlosalberto commented:
Hey @MikeGoldsmith thanks for the PR.

We can't add the dropped information to the API interfaces, as there's no need for users to know about it - we only need to expose it in the SDK side (`SpanData` and `TimedEvent` are there, for example).

MikeGoldsmith commented:
@carlosalberto I've removed from API interfaces and added directly to SpanData and nested classes.

@jkwatson I rebased my changes based on master, but now my PR seems to encompass the majority of those changes which is odd.

MikeGoldsmith commented:
Okay - I've fixed up the commit history, this is ready for review again.

jkwatson commented:
Are you planning on having the `RecordEventsReadableSpan` set this value in a later PR?

bogdandrutu commented on contrib/trace_utils/src/main/java/io/opentelemetry/contrib/trace/MessageEvent.java:
```diff
@@ -112,7 +118,17 @@ public String getName() {
+   * @return the number of dropped attributes.
+   */
+  public int getDroppedAttributeCount() {
+    return droppedAttributeCount;
```
Changes in this file are not needed.

bogdandrutu commented:
We still need to implement the logic of dropping attributes if more than configured in the TraceConfig. I marked this PR as updating the issue not fixing it.
